pkgs/racket-doc/scribblings/reference/trace.scrbl

```diff
@@ -165,6 +175,104 @@ traced call. It receives the name of the function, the function's
 ordinary arguments, its keywords, the values of the keywords, and a
 number indicating the depth of the call.
 
+Modifying this is useful to get more readable or additional output when tracing syntax transformers.
+For example, we may only care about the shape of the syntax object and not the source location
+information.
+By modifying @racket[current-trace-print-args], we can print the argument to (and, using
+@racket[current-trace-print-result], the result of) the syntax transformer as a datum using
+@racket[syntax->datum], as follows.
+
+@examples[#:eval ev
+  (require (for-syntax racket/trace))
+  (begin-for-syntax
+    (current-trace-print-args
+      (let ([ctpa (current-trace-print-args)])
+        (lambda (s l kw l2 n)
+          (ctpa s (map syntax->datum l) kw l2 n))))
+    (current-trace-print-results
+      (let ([ctpr (current-trace-print-results)])
+        (lambda (s r n)
+         (ctpr s (map syntax->datum r) n)))))
+
+  (trace-define-syntax fact
+    (syntax-rules ()
+      [(_ x) 120]))
+  (fact 5)]
+
+In this example, we update the printers by storing the current printers, @racket[ctpa] and
+@racket[ctpr], which we call after converting the arguments @racket[l] and results @racket[r] to
+datum.
+
+Note that this modification of @racket[current-trace-print-args] and
```

Comment:
Why not use `maybe-syntax->datum` in the first example, and remove the second codeblock?

(I think the note about the imperative update is useful)

Reply:
I think it's valuable to see the failure to understand how one should transform `l`.